transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -492,5 +545,232 @@ protected final void clearEpollIn0() {
                 unsafe().close(unsafe().voidPromise());
             }
         }
+
+        @Override
+        public void connect(
+                final SocketAddress remoteAddress, final SocketAddress localAddress, final ChannelPromise promise) {
+            if (!promise.setUncancellable() || !ensureOpen(promise)) {
```

Comment:
hmm I can not remember anymore why it was deprecated and there is no replacement. Maybe I should revert the `@deprecated` ?